services/server/server.go

```diff
@@ -493,6 +514,67 @@ func (s *Server) ChangeSettings(ctx context.Context, req *serverpb.ChangeSetting
 	return res, nil
 }
 
+func (s *Server) validateSSHKey(sshKey string) error {
+	tempFile := path.Join(os.TempDir(), "temp_keys")
```

Comment:
Concurrent calls to this method will overwrite the same file. We should either use a mutex, or ioutil.TempFile.